src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticAssets.targets

```diff
@@ -0,0 +1,172 @@
+      File="$(_GeneratedAspnetCoreStaticAssetInputsCacheFile)" 
+      Overwrite="True" 
+      WriteOnlyWhenDifferent="True" />
+
+    <ItemGroup>
+      <FileWrites Include="$(_GeneratedAspnetCoreStaticAssetInputsCacheFile)" />
+    </ItemGroup>
+
+  </Target>
+
+  <!-- 
+    This target generates a manifest for development time that includes information
+    about the base path for the referenced package and project static assets. The
+    manifest includes the content root and the base path for each of the referenced
+    packages and projects.
+
+    Ideally, each package/project contains a unique base path and a given content
+    root, but we don't check for duplicates on neither of them.
+     -->
+
+  <Target 
+    Name="GenerateAspNetCoreStaticAssetsManifest"
+    Inputs="$(_GeneratedAspnetCoreStaticAssetInputsCacheFile)"
+    Outputs="$(_GeneratedAspNetCoreStaticAssetsDevelopmentManifest)"
+    BeforeTargets="AssignTargetPaths"
+    DependsOnTargets="$(GenerateAspNetCoreStaticAssetsManifestDependsOn)">
+  
+    <GenerateAspNetCoreStaticAssetsManifest 
+      ContentRootDefinitions="@(_ExternalAspNetCoreStaticAssetManifestEntry)"
+      TargetManifestPath="$(_GeneratedAspNetCoreStaticAssetsDevelopmentManifest)" />
+  
+      <!-- This is the list of inputs that will be used for generating the manifest used during development. -->
+    <ItemGroup>
+      <EmbeddedResource Condition="'@(_ExternalAspNetCoreStaticAssetManifestEntry->Count())' != '0'"
+        Include="$(_GeneratedAspNetCoreStaticAssetsDevelopmentManifest)"
+        LogicalName="Microsoft.AspNetCore.StaticAssets.xml" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <FileWrites Include="$(_GeneratedAspNetCoreStaticAssetsDevelopmentManifest)" />
+    </ItemGroup>
+
+  </Target>
+
+  <!-- This target collects all the AspNetCoreStaticAsset from different sources:
+       * The current project AspNetCoreStaticAsset that come from wwwroot\** by default.
+       * Assets from referenced projects that get retrieved invoking an MSBuild target on
+         the referenced projects.
+       * Assets from the referenced packages. These will be implicitly included when nuget
+         restores the package and includes the package props file for the package.
+  -->
+  <Target 
+    Name="ResolveAspNetCoreStaticAssetsInputs">
+    <PropertyGroup>
+      <!-- The _SafeBasePath is used as a path segment in the urls that we will
+           be exposing content from when the library is referenced as a package
+           or as a project by a web application. Our convention will be to expose
+           content directly on _content/<<_SafeBasePath>>
+           
+           We simply remove the dots from the package id so that a package
+           like Microsoft.AspNetCore.Identity becomes MicrosoftAspNetCoreIdentity 
+           
+           TODO: Investigate if we need to do something more sophisticated here.
+      -->
+      <_SafeBasePath>$(PackageId.Replace('.',''))</_SafeBasePath>
```

Comment:
This I'm going to change in the next PR for an MSBuild task. The idea is that we use your PackageId and produce a default name from it that does not contain problematic symbols for the url. (`.` while allowed, are problematic)

This will be our default convention and you'll be able to change it.

Reply:
FWIW I don't think this needs to be customizable.